handler/authentication_handler.go

```diff
@@ -28,28 +35,65 @@ func LoginHandler(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, CreateResponse(fmt.Sprintf("%v", err)))
 		return
 	}
+	// get compnay name
+	providerDetails, err := provider.ProviderControllerObj.GetProvider(credentials.Email)
+	if err != nil {
+		logrus.WithField("err", err).Error("error getting provider")
+		c.JSON(http.StatusBadRequest, CreateResponse(fmt.Sprintf("%v", err)))
+		return
+	}
+	resp.CompanyName = providerDetails.CompanyName
+
 	c.JSON(http.StatusOK, resp)
 	return
 }
 
 func SignUpHandler(c *gin.Context) {
 	var (
-		credentials dto.Credentials
+		signUpRequest   SignUpRequest
+		credentials     dto.Credentials
+		providerDetails dto.Provider
 	)
 
-	err := c.BindJSON(&credentials)
+	err := c.BindJSON(&signUpRequest)
 	if err != nil {
 		logrus.WithField("err", err).Error("error params")
 		c.JSON(http.StatusBadRequest, CreateResponse(fmt.Sprintf("%v", err)))
 		return
 	}
+	// check if company name already exist
+	if provider.ProviderControllerObj.IsCompanyExist(signUpRequest.CompanyName) {
```

Comment:
@sweiyang , this scenario means each provider user requires a unique company. You cannot have 2 employees of the same company to sign up.

Reply:
nope cannot. The first provider that signs up as a new company will be a super admin. Afterwards, we will have a sections for super admin to create employee account to access